### PR TITLE
Cherry-pick #12816 to 7.2: [Metricbeat] Avoid omitting of errors in Vsphere module and upgrade to use context.Context

### PR DESCRIPTION
**Pull request by sayden**
Cherry-pick of PR #12816 to 7.2 branch. Original message: 

Some potential error checks are being omitted in vSphere module. They aren't probably key to fetch metrics but they may be hiding important debug information when things go wrong.

This PR prints in debug level if any of those errors have returned information.

At the same time, the PR is upgraded to use `context.Context`in Fetch method, recently added to Beats and it may be useful as a background context was being used on the Metricsets.

### CODE REVIEW

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -41,6 +41,30 @@ https://github.com/elastic/beats/compare/v7.2.0...7.2[Check the HEAD diff]
 *Metricbeat*
 
 - Fix connections leak in redis module {pull}12914[12914] {pull}12950[12950]
+- Change diskio metrics retrieval method (only for Windows) from wmi query to DeviceIOControl function using the IOCTL_DISK_PERFORMANCE control code {pull}11635[11635]
```
Comment (kaiyan-sheng) [inline CHANGELOG.next.asciidoc]: oops so many changelog entries!

Comment (sayden) [reply CHANGELOG.next.asciidoc]: oink! :pig: so dirty! Thanks! :smile:

File: metricbeat/module/vsphere/datastore/datastore_test.go [hunk 2]
```diff
@@ -85,9 +85,9 @@ func TestData(t *testing.T) {
 	ts := model.Service.NewServer()
 	defer ts.Close()
 
-	f := mbtest.NewReportingMetricSetV2Error(t, getConfig(ts))
+	f := mbtest.NewReportingMetricSetV2WithContext(t, getConfig(ts))
 
-	if err := mbtest.WriteEventsReporterV2Error(f, t, ""); err != nil {
+	if err := mbtest.WriteEventsReporterV2WithContext(f, t, ""); err != nil {
```
Comment (kaiyan-sheng) [inline metricbeat/module/vsphere/datastore/datastore_test.go]: `WriteEventsReporterV2WithContext` seems to be undefined for 7.2, maybe we should keep `WriteEventsReporterV2Error` here?

Comment (sayden) [reply metricbeat/module/vsphere/datastore/datastore_test.go]: :man_facepalming: Yeap, good catch :grimacing:

**Comment by sayden** [comment]
Error seems unrelated, merging

**Pull request merged**